app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,308 @@
+            },
+        }, componentId);
+
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+
+        return () => {
+            unsubscribe.remove();
+            backHandler.remove();
+        };
+    }, [userInfo, isTablet, profileImage]);
+
+    useEffect(() => {
+        if (!isTablet) {
+            setButtons(componentId, {rightButtons: [rightButton]});
+        }
+    }, [isTablet]);
+
+    const style = getStyleSheet(theme);
+    const service = currentUser.authService;
+
+    const rightButton = {
+        id: 'update-profile',
+        enabled: false,
+        showAsAction: 'always' as const,
+        testID: 'edit_profile.save.button',
+        color: theme.sidebarHeaderTextColor,
+        text: intl.formatMessage({id: 'mobile.account.settings.save', defaultMessage: 'Save'}),
+    };
+
+    const close = () => {
+        if (isModal) {
+            dismissModal({componentId});
+        } else if (isTablet) {
+            DeviceEventEmitter.emit(Events.ACCOUNT_SELECT_TABLET_VIEW, '');
+        } else {
+            popTopScreen(componentId);
+        }
+
+        return true;
+    };
+
+    const enableSaveButton = (value: boolean) => {
+        if (isTablet) {
+            setCanSave(value);
+        } else {
+            const buttons = {
+                rightButtons: [{...rightButton, enabled: value}],
+            };
+
+            setButtons(componentId, buttons);
+        }
+    };
+
+    const submitUser = async () => {
+        enableSaveButton(false);
+        setError(undefined);
+        setUpdating(true);
+        try {
+            const partialUser: Partial<UserProfile> = {
+                email: userInfo.email,
+                first_name: userInfo.firstName,
+                last_name: userInfo.lastName,
+                nickname: userInfo.nickname,
+                position: userInfo.position,
+                username: userInfo.username,
+            };
+
+            const {error: reqError} = await updateMe(serverUrl, partialUser);
+
+            if (reqError) {
+                return resetScreen(reqError as Error);
+            }
+
+            close();
+        } catch (e) {
+            return resetScreen(e as Error);
+        }
+
+        return null;
+    };
+
+    const resetScreen = (resetError: Error) => {
+        setError(resetError?.message);
+        setUpdating(false);
+        enableSaveButton(true);
+        scrollViewRef.current?.scrollToPosition(0, 0, true);
+    };
+
+    const updateField = useCallback((id: string, name: string) => {
+        const update = {...userInfo};
+        update[id] = name;
+        setUserInfo(update);
+
+        // @ts-expect-error access object property by string key
+        const currentValue = currentUser[id];
+        const didChange = currentValue !== name;
+        enableSaveButton(didChange);
+    }, [userInfo]);
+
+    const setScrollViewRef = useCallback((ref) => {
+        scrollViewRef.current = ref;
+    }, []);
```

Comment:
huh? you should be using `useRef` instead.

Reply:
Do you need my help with this?

Reply:
Refer to server screen for this part

Reply:
Done